### PR TITLE
Suggest `log` extension for use in Zed log view

### PR DESCRIPTION
**Pull request by d1y**
We now have a log-extension in the extension. This PR will automatically highlight the log when you install the log-extension
> The extension repo is https://github.com/evrsen/zed-log

before

Release Notes:

- Improved zed:OpenLog

### CODE REVIEW

File: crates/zed/src/zed.rs [hunk 1]
```diff
@@ -572,6 +575,14 @@ fn open_log_file(workspace: &mut Workspace, cx: &mut ViewContext<Workspace>) {
                             .update(cx, |project, cx| project.create_buffer(&log, None, cx))
                             .expect("creating buffers on a local workspace always succeeds");
 
+                        // If the log language exists, set the buffer to the log language
+                        // The log language repo is https://github.com/evrsen/zed-log
+                        if let Ok(log) = logLang {
```
Comment (SomeoneToIgnore) [inline crates/zed/src/zed.rs:580]: > `logLang` 

Huh, clippy seems to be fine with non-snake-cased namings?
Here's a small patch on top if you want to consider a somewhat better styling of the code:

```
diff --git a/crates/zed/src/zed.rs b/crates/zed/src/zed.rs
index 51959ede3..933ff703a 100644
--- a/crates/zed/src/zed.rs
+++ b/crates/zed/src/zed.rs
@@ -529,9 +529,15 @@ fn open_log_file(workspace: &mut Workspace, cx: &mut ViewContext<Workspace>) {
         .with_local_workspace(cx, move |workspace, cx| {
             let fs = workspace.app_state().fs.clone();
             let app_state = workspace.app_state().clone();
+            // `"LOG"` is taken from [1] as the current "official" log language plugin
+            // [1]: https://github.com/evrsen/zed-log/blob/b913d7174675ec5efde945b8fb8f3e09811d8d28/languages/log/config.toml#L1
+            let log_language_name = "LOG";
             cx.spawn(|workspace, mut cx| async move {
-                let (old_log, new_log) =
-                    futures::join!(fs.load(&paths::OLD_LOG), fs.load(&paths::LOG));
+                let (old_log, new_log, log_language) = futures::join!(
+                    fs.load(&paths::OLD_LOG),
+                    fs.load(&paths::LOG),
+                    app_state.languages.language_for_name(log_language_name),
+                );
                 let log = match (old_log, new_log) {
                     (Err(_), Err(_)) => None,
                     (old_log, new_log) => {
@@ -555,8 +561,6 @@ fn open_log_file(workspace: &mut Workspace, cx: &mut ViewContext<Workspace>) {
                     }
                 };
 
-                let logLang = app_state.languages.language_for_name("LOG").await;
-
                 workspace
                     .update(&mut cx, |workspace, cx| {
                         let Some(log) = log else {
@@ -572,17 +576,11 @@ fn open_log_file(workspace: &mut Workspace, cx: &mut ViewContext<Workspace>) {
                         };
                         let project = workspace.project().clone();
                         let buffer = project
-                            .update(cx, |project, cx| project.create_buffer(&log, None, cx))
+                            .update(cx, |project, cx| {
+                                project.create_buffer(&log, log_language.ok(), cx)
+                            })
                             .expect("creating buffers on a local workspace always succeeds");
 
-                        // If the log language exists, set the buffer to the log language
-                        // The log language repo is https://github.com/evrsen/zed-log
-                        if let Ok(log) = logLang {
-                            buffer.update(cx, |buffer, cx| {
-                                buffer.set_language(Some(log), cx);
-                            });
-                        }
-
                         let buffer = cx.new_model(|cx| {
                             MultiBuffer::singleton(buffer, cx).with_title("Log".into())
                         });
```

**Review by SomeoneToIgnore — COMMENTED** [review]
Thank you, it's a nice idea overall, but I am not happy with the `"LOG"` hardcode.

I'm not familiar with that corner well, but we seem to have autosuggestions: https://github.com/zed-industries/zed/pull/9941

Can we instead alter the buffer or suggestions somehow, so that log buffer being opened is instead suggested to have a log extension used?
This way we won't have any hardcode in the zed.rs at all, as `LOG` could change to `log` or `loglang` or whatnot for other extensions — but the suggestions is a contract that should work nonetheless.

**Review by SomeoneToIgnore — COMMENTED** [review]
Thank you for applying the suggestion, now, to the last part: can we avoid using `app_state.languages.language_for_name` at all and do the extension suggestions to do the work, as in https://github.com/zed-industries/zed/pull/9941 ?

**Comment by d1y** [comment]
> Thank you for applying the suggestion, now, to the last part: can we avoid using `app_state.languages.language_for_name` at all and do the extension suggestions to do the work, as in #9941 ?

So the best thing to do is?:
1.  zed:OpenLog set as file-extenstion (1.log)
2. `extension_suggest.rs` -> `SUGGESTED.get_or_init` add ( `log` | `log`)

**Comment by SomeoneToIgnore** [comment]
Yes, this way we'll keep all extension-related knowledge in the same place with the suggestions, while the current PR's state adds an extra `"LOG"` hardcode and `.await` inside the action workflow.

**Comment by d1y** [comment]
> Yes, this way we'll keep all extension-related knowledge in the same place with the suggestions, while the current PR's state adds an extra `"LOG"` hardcode and `.await` inside the action workflow.

Yep. But now I'm stuck on the first step: **How to set file-extenstion?**

--------------------------------------------------------------

1. Create real .log tmp file(write `let log = match (old_log, new_log) {` with `log` content)
2. Open real .log file in workspace(editor)

> The tmp file it will be deleted automatically? (golang os.Temp)

**Comment by SomeoneToIgnore** [comment]
I wonder if we need the old log at all: log "rotation" happens only on 1MiB threshold:
https://github.com/zed-industries/zed/blob/1360dffeadc7367422a4f4802da75d68ce57de59/crates/zed/src/main.rs#L523-L526
and that's usually more than enough for people to look through for pasting into bug reports.

So we can open the log file directly, and ignore the old one entirely, that should be easy to open with the existing API and should trigger the corresponding plugin installation prompt due to the `*.log` extension?

**Comment by SomeoneToIgnore** [comment]
Closing as it's not moving anywhere, but it's a good idea overall, we should merge something like this without hardcoding the language.

**Comment by d1y** [comment]
> So we can open the log file directly, and ignore the old one entirely

Can we do this now? We just need to open the real log file(`paths::LOG`)

**Comment by SomeoneToIgnore** [comment]
We can, of course: just have to add the log extension association.

**Comment by SomeoneToIgnore** [comment]
> Is there any progress?

From a person who have a least watched this repository, any educated guess?

>  Can we highlight the log to the zed core code level so that things can be easier to handle

Assuming that the idea is to move the log language and grammar into Zed — could work, but as discussed over a year ago in this issue, we'd better have a way to say which extension is required for a certain buffer?
At least emulate this with an extension somehow.

After knowing the idea of a better approach, waiting for over a year to induce the dirty approach is not looking superb.

**Pull request closed without merging**